src/api/app/controllers/webui/repositories_controller.rb

```diff
@@ -148,14 +148,16 @@ def create_dod_repository
   # POST project/create_image_repository
   def create_image_repository
     authorize @project, :update?
-    repository = @project.repositories.create!(name: 'images')
-    Architecture.available.each do |architecture|
-      repository.repository_architectures.create!(architecture: architecture)
-    end
-    @project.prepend_kiwi_config
+    repository = @project.repositories.new(name: 'images')
 
     if repository.save
+      Architecture.available.each do |architecture|
+        repository.repository_architectures.create!(architecture: architecture)
+      end
+
+      @project.prepend_kiwi_config
       @project.store
+
```

Comment:
This line was added because I found it more readable. If someone is strongly against this, I can remove it again, of course.